drivers/flash/spi_nor.c

```diff
@@ -1024,8 +1024,12 @@ static int spi_nor_configure(const struct device *dev)
 		return -ENODEV;
 	}
 
-	/* Might be in DPD if system restarted without power cycle. */
-	exit_dpd(dev);
+	/* After a soft-reset the flash might still be busy writing/erasing or in DPD.
+	 * Exit DPD and wait until ready.
+	 */
+	acquire_device(dev);
+	spi_nor_wait_until_ready(dev);
```

Comment:
When it is needed to wait for the flash, it would be nice to let the user know what is going on. MISO could be also constantly high, e.g. because of some wrong pin configuration. It would not be good to just silently wait here forever in such case.
How about something like:
```suggestion
	rc = spi_nor_rdsr(dev);
	if (rc > 0 && (rc & SPI_NOR_WIP_BIT)) {
		LOG_WRN("Waiting until flash is ready");
		spi_nor_wait_until_ready(dev);
	}
```

Reply:
I've added your suggestions, but I wasn't able to configure the logging so the warning is output at the right time. With `CONFIG_LOG_MODE_DEFERRED` the warning isn't output until after the erasing is done, and with `CONFIG_LOG_MODE_IMMEDIATE` it looks like the warning is printed before `*** Booting Zephyr OS build zephyr-v3.2.0-...  ***`. Probably because of the mix of `printk` and `LOG_XXX`. Maybe I've configured the logging incorrectly?

```
CONFIG_LOG=y
CONFIG_LOG_MODE_IMMEDIATE=y
CONFIG_FLASH_LOG_LEVEL_DBG=y
```

Reply:
Initialization of the driver is done before the banner is printed, so it is normal that for immediate logging you get the message before the banner.
For deferred logging, adding the following right after `LOG_WRN()` should help:
```
		while (LOG_PROCESS()) {
		}
```